### ISSUE TITLE
Addition of Faction Flags/Emblems

### ISSUE THREAD

**Issue by Janaszar** [endless-sky/endless-sky#5028]
I know there has been some talk about developing faction emblems, mainly for licenses.  What if we used those emblems/flags when we target a ship?  For example, 
![Untitled](https://user-images.githubusercontent.com/32232509/80771823-9d27c680-8b22-11ea-961d-1165817a2da7.png)

I think it would be nice to see these rarely displayed flags/emblems more often.

**Describe the solution you'd like**
I've already implemented it in another fork of ES: <https://github.com/kikotheexile/Endless-Sky-Civil-War/pull/72/files>
Code works, compiles fine, and the images are just placeholders.

Mainly putting this as an Issue/Feature Request as I'd like some visibility before I push anything to source code single-handedly here.  (I'm also sure there are a few things that could be cleaned up in the code)
I also need the appropriate emblems/faction flags, from whoever made them?  I know I've seen them somewhere for ES, but can't remember any time recently.  Well, aside from the licenses.

**Comment by Amazinite** [comment]
I don't think that having an emblem show up when you target a ship is the best way to show such visuals off. For one, it seems to me like it'd just be unnecessary UI clutter. I think that a far better place for flags/emblems would probably be in the logbook, using the feature requested in #4725.

**Comment by Zitchas** [comment]
Personally, I like the idea of having a symbol / logo for each faction and have it display under the targeting reticule. That being said, I also think that it should have an on/off setting so that people aren't forced to have it. I'd opt to have it visible, though.

**Comment by beccabunny** [comment]
How about putting it in the hail panel instead? Atm it has quite a lot of space, and with that + using the thumbnails in hails it would become very interesting to use

**Comment by Zitchas** [comment]
Eh, in the hail panel I'll open it up once or twice just to see it; then never see it again. I basically only use the "hail" function to demand tribute from planets - and I don't do that much, so that means I basically never use the hail. So I'd never see these symbols. 

Actually, it occurs to me that putting these symbols beneath the targeting reticle actually serves another purpose: Accessibility. There are more than a few people in the world who have some variation of color-blindness. Having a distinct symbol there to identify faction (which could be easier to recognize peripherally than the actual label) could make the game more accessible for people who have trouble with some colors.

Also, if the boarding system worked a bit differently it could be used there, too. For instance, if instead of the current board / assist differentiation, what if both just led to the same boarding screen, and instead of having the option to loot, attack, defend, or leave; there was also the option to repair or offload cargo? For that matter, having a hailing thing come up alongside the boarding would be interesting, because then when you board you could have things happen like, say, the captain of the ship offers you X credits if you repair the ship instead of looting things. Or a mission objective could be to transfer Y cargo to a ship that is waiting in Z system. In all cases, having the faction that the other ship is a part of represented on the screen could be useful or simply an interesting detail.

**Comment by tehhowch** [comment]
What can be done is support drawing this emblem in the respective interfaces when the interface specifies it. Then plugins could add (or remove) it to the targeting hud, or hailpanel, as desired

**Comment by Zitchas** [comment]
Of course, then there's the other problem of needing to make symbols for each faction... 

Merchants probably should have an icon, although it should probably be something fairly simple. Think of it more as a "symbol for independent entrepreneurs" rather than an actual "This is our Logo!" sort of thing.

**Comment by kikotheexile** [comment]
This is something that has already been merged into ESS, it doesn't really clutter the UI at all, instead it fills a weird gap on the left hand section of the screen, (Example image below using a recolored placeholder image)  and provides a generalist symbol. For our faction flags we are using emblems created by Bluebolt (Alophox#3346) and one created by bluebolt but modified @jafdy for testing. 

Faction emblems *should* be incredibly simple, for example, for Merchants I am planning on using a resized version MZ's Endless Sky compass. I also did some testing using full square backgrounds which... honestly look like crap. This is absolutely more of an emblem feature and less of a flag. 

I feel this would work beautifully in concert with #4725 

(example image)
![free worlds UI clutter](https://user-images.githubusercontent.com/19562488/80893148-6cb46980-8c95-11ea-9583-eab9e04054f2.png)

**Comment by Zitchas** [comment]
Your screen seems... Short, for some reason.

But either way, the symbol looks good. Nice to see a practical example of it working. The ES main screen logo is a good idea for the merchant logo.

**Comment by Zireael07** [comment]
@kikotheexile: Is ESS the same as Endless-Sky-Civil-War?

**Comment by kikotheexile** [comment]
> @kikotheexile: Is ESS the same as Endless-Sky-Civil-War?
ESS is the engine, Endless Sky Civil War is the content. But for now they share a developing github.
